From: Roland Fehrenbacher <rf@q-leap.de>
Date: Thu, 26 Nov 2015 12:27:08 +0100
Subject: QL: Prevent memset error for older gcc (as in trusty)

---
 psm_diags.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/psm_diags.c b/psm_diags.c
index 90f2e17..4552016 100644
--- a/psm_diags.c
+++ b/psm_diags.c
@@ -32,6 +32,7 @@
  * SOFTWARE.
  */
 
+#include <assert.h>
 #include "psm_user.h"
 #include "psm_mq_internal.h"
 
@@ -244,6 +245,7 @@ void *memcpy_check_one (memcpy_fn_t fn, void *dst, void *src, size_t n)
 	  ((uintptr_t) dst ^ (uintptr_t) src ^ (uintptr_t) n);
   unsigned int state;
   size_t i;
+  assert (n > 0);
   memset(src, 0x55, n);
   memset(dst, 0xaa, n);
   srand(seed);
